crates/bevy_ecs/src/reflect/map_entities.rs

```diff
@@ -12,53 +14,73 @@ use bevy_reflect::FromType;
 /// See [`SceneEntityMapper`] and [`MapEntities`] for more information.
 #[derive(Clone)]
 pub struct ReflectMapEntities {
-    map_all_entities: fn(&mut World, &mut SceneEntityMapper),
-    map_entities: fn(&mut World, &mut SceneEntityMapper, &[Entity]),
+    map_entities: fn(&mut dyn Reflect, &mut dyn DynEntityMapper),
+    map_all_world_entities: fn(&mut World, &mut SceneEntityMapper),
+    map_world_entities: fn(&mut World, &mut SceneEntityMapper, &[Entity]),
 }
 
 impl ReflectMapEntities {
+    /// A general method for applying [`MapEntities`] behavior to a reflected component.
+    ///
+    /// Be mindful in its usage: Works best in situations where the entities in the [`EntityHashMap<Entity>`] are newly
+    /// created, before systems have a chance to add new components.
```

Comment:
Maybe briefly explain why this is?

Reply:
Mmmm, might not actually be relevant. I copied the original comment and removed the definitely-not-relevant bits, but this maybe should've made that cut as well.

Reply:
OK, thought about it some more and definitely not relevant. Replaced with a new warning advising callers to ensure that entities already exist in advance.